src/test/java/org/springframework/data/hazelcast/repository/query/HazelcastSortAccessorTest.java

```diff
@@ -0,0 +1,146 @@
+package org.springframework.data.hazelcast.repository.query;
+
+import com.hazelcast.query.Predicate;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.springframework.data.domain.Sort;
+import org.springframework.data.history.RevisionSort;
+import org.springframework.data.keyvalue.core.query.KeyValueQuery;
+import org.springframework.util.ObjectUtils;
+
+import java.io.Serializable;
+import java.util.*;
+
+import static org.junit.Assert.*;
+
+public class HazelcastSortAccessorTest {
+
+    @Rule
+    public ExpectedException exceptionRule = ExpectedException.none();
+
+    @Test()
+    public void test1() {
+        exceptionRule.expect(UnsupportedOperationException.class);
+        exceptionRule.expectMessage("Null handling not implemented:");
+        KeyValueQuery<String> keyValueQuery0 = new KeyValueQuery<String>("^pg~]=P");
+        Sort.Order[] sort_OrderArray0 = new Sort.Order[7];
```

Comment:
Tests are quite unreadable at the moment for a person that sees the code for the first time - you could improve that by introducing a clear _given/when/then_ structure to them

Reply:
thank you, I have introduced proper given/when/then structure to them now.